test/e2e/scheduling/predicates.go

```diff
@@ -900,16 +892,38 @@ func getRequestedCPU(pod v1.Pod) int64 {
 	return result
 }
 
-func waitForScheduler() {
-	// Wait a bit to allow scheduler to do its thing
-	// TODO: this is brittle; there's no guarantee the scheduler will have run in 10 seconds.
-	framework.Logf("Sleeping 10 seconds and crossing our fingers that scheduler will run in that time.")
-	time.Sleep(10 * time.Second)
+// removeTaintFromNodeAction returns a closure that removes the given taint
+// from the given node upon invocation.
+func removeTaintFromNodeAction(cs clientset.Interface, nodeName string, testTaint v1.Taint) func() error {
+	return func() error {
+		framework.RemoveTaintOffNode(cs, nodeName, testTaint)
```

Comment:
The `action` that `common.ObserveEventAfterAction` takes must be of type 'func() error'. That's why this closure has that type. However, the method does error checking internally and does not return any error. I could rewrite another method similar to `framework.RemoveTaintOffNodes` that returns an error, but it would be mostly duplicate code.